orttraining/tools/scripts/pipeline_model_split.py

```diff
@@ -0,0 +1,376 @@
+import sys
+import onnx
+from onnx import helper
+from onnx import TensorProto
+from onnx import OperatorSetIdProto
+
+# Edge that needs to be cut for the split.
+# If the edge is feeding into more than one nodes, and not all the nodes belong to the same cut,
+# specify those consuming nodes that need to be cut
+class CutEdge:
+    def __init__(self, edgeId, consumingNodes=None):
+        self.edgeId = edgeId
+        self.consumingNodes = consumingNodes
+
+def add_expand_shape(model, name, shape):
+    expand_edge = model.graph.value_info.add()
+    expand_edge.name = name
+    expand_edge.type.CopyFrom(shape)
```

Comment:
>type.CopyFrom(shape [](start = 16, length = 19)

seems not right, param says shape, but actually is copying type..

Reply:
yes, this is a bit confusing. The data structure from value_info is using name 'type', but it actually contains shape information too. 

But this could be too detail. I'll update the name such that it is consistant in the script. 

---
In reply to: [400683688](https://github.com/microsoft/onnxruntime/pull/3275#discussion_r400683688) [](ancestors = 400683688)